### PR TITLE
Update documentation and add choices for status parameter

### PR DESCRIPTION
**Pull request by ilicmilan**
##### SUMMARY
Update documentation and add choices for status parameter

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME

- **gcp_compute_instance**

### CODE REVIEW

File: lib/ansible/modules/cloud/google/gcp_compute_instance.py [hunk 1]
```diff
@@ -434,13 +434,13 @@
         type: bool
   status:
     description:
-    - 'The status of the instance. One of the following values: PROVISIONING, STAGING,
-      RUNNING, STOPPING, SUSPENDING, SUSPENDED, and TERMINATED.'
-    - As a user, use RUNNING to keep a machine "on" and TERMINATED to turn a machine
-      off .
-    - 'Some valid choices include: "PROVISIONING", "STAGING", "RUNNING", "STOPPING",
-      "SUSPENDING", "SUSPENDED", "TERMINATED"'
+    - 'The desired state of the instance. One of the following values:
```
Comment (samccann) [inline lib/ansible/modules/cloud/google/gcp_compute_instance.py]:
```suggestion
    - 'The desired state of the instance.
```

File: lib/ansible/modules/cloud/google/gcp_compute_instance.py [hunk 2]
```diff
@@ -434,13 +434,13 @@
...
+      RUNNING, STOPPED, SUSPENDED, and TERMINATED.'
```
Comment (samccann) [inline lib/ansible/modules/cloud/google/gcp_compute_instance.py]:
```suggestion
```

File: lib/ansible/modules/cloud/google/gcp_compute_instance.py [hunk 3]
```diff
@@ -434,13 +434,13 @@
...
+    - As a user, use RUNNING to keep a machine "on" and TERMINATED or STOPPED
+      to turn a machine off .
+    - 'Some valid choices include: "RUNNING", "STOPPED", "SUSPENDED", "TERMINATED"'
```
Comment (samccann) [inline lib/ansible/modules/cloud/google/gcp_compute_instance.py]:
```suggestion
```

Comment (samccann) [reply lib/ansible/modules/cloud/google/gcp_compute_instance.py]: These states are all repeated under choices so don't need to add them here except for where you are explaining what each means.

Comment (ilicmilan) [reply lib/ansible/modules/cloud/google/gcp_compute_instance.py]: Agree. Fixed

**Review by samccann — APPROVED** [review]
Docs portion LGTM

**Comment by ilicmilan** [comment]
What's going on with this PR?
Are we gonna merge it? :smiley:

**Comment by samccann** [comment]
Anything with code has to be merged by a developer. Please consider bringing this up at the core IRC meeting this week to help nudge it along.

**Comment by ilicmilan** [comment]
@erjohnso @rambleraptor

**Comment by ilicmilan** [comment]
Hi @erjohnso , could you please check this one?

**Comment by rambleraptor** [comment]
Hello!

We've discussed putting values into the 'choices' field before and we've decided against it. Our biggest worry is that the GCP API enables a new value and it doesn't work with our modules because of a now-faulty client-side check.

By not using the 'choices' field, we leave validation up to the GCP API, which will always be more correct and up-to-date than our Ansible modules.

**Comment by ilicmilan** [comment]
Hello!
Thanks for the response! :)
That makes sense.
Does it mean other changes are fine?

**Comment by ilicmilan** [comment]
Here it is.
There is another pull request I would like to create which relies on this change.

**Comment by ilicmilan** [comment]
I've opened another PR which includes these changes.
https://github.com/ansible/ansible/pull/63465

**Pull request closed without merging**